src/thor/map_matcher.cc

```diff
@@ -241,11 +247,11 @@ MapMatcher::FormPath(meili::MapMatcher* matcher,
     const GraphTile* end_tile = matcher->graphreader().GetGraphTile(prior_node);
     nodeinfo = end_tile->node(prior_node);
 
-    // Create a predecessor EdgeLabel (for transition costing)
-    pred = {kInvalidLabel, edge_id, directededge, {}, 0, 0, mode, 0};
+    // Update the predecessor EdgeLabel (for transition costing in the next round);
+    pred = {kInvalidLabel, edge_id, directededge, elapsed, 0, 0, mode, 0};
 
     // Add to the PathInfo
-    path.emplace_back(mode, elapsed.secs, edge_id, 0, elapsed.cost, false);
+    path.emplace_back(mode, elapsed.secs, edge_id, 0, elapsed.cost, false, turn_cost.secs);
```

Comment:
here we store the turn cost at the beginning of the edge so that on the outside we can subtract it off